### DISCUSSION TITLE
Internal Server Error after changing Engine preferences through web GUI

### DISCUSSION THREAD

**Discussion by js4github** [searx/searx#3523] [Installation issue: custom]
**Solution:** Increasing the uwsgi buffer-size to 5120 seems to have corrected the issue for me. Details and diffs are below.
[Reference: FreedomBox forum support post](https://discuss.freedombox.org/t/solved-searx-500-internal-server-error/2567/4)
[Reference: FreedomBox forum development post](https://discuss.freedombox.org/t/searx-correct-500-errors-when-setting-preferences-from-the-web-ui/2574)

**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**
Web UI version info
>Powered by [searx](https://searx.github.io/searx) - 0.18.0 - a privacy-respecting, hackable metasearch engine

Debian package info
>ii  python3-searx                        0.18.0+dfsg1-1                        all          Privacy-respecting metasearch engine - library package
>ii  searx                                0.18.0+dfsg1-1                        all          Privacy-respecting metasearch engine

**Fork:** I'm unsure if I'm using a fork

**How did you install Searx?**
Searx was installed using Debian / Freedombox automation
**OS Info**
> PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
> NAME="Debian GNU/Linux"
> VERSION_ID="11"
> VERSION="11 (bullseye)"
> VERSION_CODENAME=bullseye
> ID=debian
> HOME_URL="https://www.debian.org/"
> SUPPORT_URL="https://www.debian.org/support"
> BUG_REPORT_URL="https://bugs.debian.org/"

**FreedomBox Info**
> You are running Debian GNU/Linux 11 (bullseye) and FreedomBox version 23.6.1. FreedomBox is up to date.

**What happened?**
After chaning the search engine options I would in some cases receive an Internal Server Error in the web UI which could only be resolved by deleting browser cookies which contained the new searx client options. Any favorable changes made up to this point are lost when the cookie set by searx is deleted.

> # Internal Server Error
>The server encountered an internal error or misconfiguration and was unable to complete your request.
>
>Please contact the server administrator at webmaster@localhost to inform them of the time this error occurred, and the actions you >performed just before this error.
>
>More information about this error may be available in the server error log.
>
>Apache/2.4.56 (Debian) Server at ...

**How To Reproduce**
1. Go to the searx web page
2. Select Preferences in the upper right 
3. Select Engines in preferences
4. Select videos in the engine category tabs
5. Select "Allow All"
6. Save preferences
7. I would see the Internal Server Error above
As an alternative to steps 4-6 you may need to "Allow All" for each category of search engine, but just doing this with videos was reliable for me.

**Expected behavior**
I expected to be returned to the searx UI after saving preferences.

**Screenshots & Logs**
[Screen capture of error](https://discuss.freedombox.org/uploads/default/original/2X/8/8fb453ef41300ed78b8f87545294820ac88272d1.png)

Error in /var/log/uwsgi/app/searx.log:
> Sun Apr 30 20:10:00 2023 - invalid request block size: 4612 (max 4096)…skip

**Additional context**
I believe that the default uwsgi buffer size is set too small for some requests which searx generates. I have been able to resolve the problem in my own case by modifying /etc/uwsgi/apps-available/searx.ini adding a line:
> buffer-size = 5120 

I selected 5120 by adding 25% to the skipped request size. After making this change I have not encountered the problem again. I did not have success setting a larger UWSGI_BUFFER_SIZE=8192 environment variable in the uwsgi startup script.

In my own mind I have confirmed success by using the "Allow All" in every server engine category and successfully saving the change and retruning to the searx web UI to perform a successful search.

**Diffs**
>===================================================================
>RCS file: RCS/searx.ini,v
>retrieving revision 1.1
>diff -r1.1 searx.ini
>11a12,14
>\> # increase buffer size from 4096 for larger searx requests
>\> buffer-size = 5120
>\>

**Comment by js4github** [comment]
Thanks, @unixfox for acting on this. I'll keep an eye on this - if there's anything I can provide just let me know.

**Comment by unixfox** [comment]
searx 0.18.0 is completely outdated, 1.1.0 came out almost one year ago: https://github.com/searx/searx/releases/tag/v1.1.0

**Reply by js4github** [reply]
Debian stable (bullseye) users who take the release upgrade to bookworm will receive searx 1.1.0.
Debian stable users who choose not to do the release upgrade and remain on the 0.18 version distributed with Debian bullseye can apply this work around to eliminate the internal server error when changing engine preferences.
